File: docs/public/specs/esm/sorted-bars.js
import * as vg from "@uwdata/vgplot";

await vg.coordinator().exec(
  vg.loadParquet("athletes", "data/athletes.parquet")
);

const $query = vg.Selection.intersect();

export default vg.vconcat(
  vg.menu({ label: "Sport", as: $query, from: "athletes", column: "sport", value: "aquatics" }),
  vg.vspace(10),
  vg.plot(
    vg.barX(
      vg.from("athletes", { filterBy: $query }),
      { x: vg.sum("gold"), y: "nationality", fill: "steelblue", sort: {"y":"-x","limit":10} }
    ),
    vg.xLabel("Gold Medals"),
    vg.yLabel("Nationality"),
    vg.yLabelAnchor("top"),
    vg.marginTop(15)
  )
);

File: docs/public/specs/python/sorted-bars.py
import vgplot as vg

athletes = vg.parquet("data/athletes.parquet")

query = vg.selection.intersect()

view = vg.vconcat(
    vg.menu(
        label="Sport", bind=query, source=athletes, column="sport", value="aquatics"
    ),
    vg.vspace(10),
    vg.plot(
        vg.bar_x(
            data=athletes,
            filter_by=query,
            x=vg.sum("gold"),
            y="nationality",
            fill="steelblue",
            sort=vg.sort(y="-x", limit=10),
        ),
        vg.x_label("Gold Medals"),
        vg.y_label("Nationality"),
        vg.y_label_anchor("top"),
        vg.margin_top(15),
    ),
)
